controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -78,6 +78,12 @@ func (w *Workload) UpdateCoreDNS(ctx context.Context, kcp *controlplanev1.Kubead
 		return nil
 	}
 
+	// If we haven't been asked to do anything, return early
+	if clusterConfig.DNS.ImageRepository == "" && clusterConfig.ImageRepository == "" &&
```

Comment:
I think we should return also if clusterConfig.DNS.Type != CoreDNS
@vincepri is it written somewhere that KCP does not support kube-dns as a DNS option?

Reply:
This is already the case, on line 77

Reply:
Thanks for pointing out